appscale/tools/agents/azure_agent.py

```diff
@@ -334,25 +413,34 @@ def create_virtual_machine(self, credentials, network_client, network_id,
                      create_option=DiskCreateOptionTypes.from_image,
                      name=vm_network_name, vhd=virtual_hd, image=image_hd)
 
-    compute_client.virtual_machines.create_or_update(
-      resource_group, vm_network_name, VirtualMachine(location=zone,
-                                                      os_profile=os_profile,
-                                                      hardware_profile=hardware_profile,
-                                                      network_profile=network_profile,
-                                                      storage_profile=StorageProfile(
-                                                        os_disk=os_disk)))
+    result = compute_client.virtual_machines.create_or_update(
+      resource_group, vm_network_name,
+      VirtualMachine(location=zone, os_profile=os_profile,
+                     hardware_profile=hardware_profile,
+                     network_profile=network_profile,
+                     storage_profile=StorageProfile(os_disk=os_disk)))
+
+    self.sleep_until_update_operation_done(result, vm_network_name,
+                                           verbose)
 
     # Sleep until an IP address gets associated with the VM.
-    while True:
-      public_ip_address = network_client.public_ip_addresses.get(resource_group,
-                                                                 vm_network_name)
-      if public_ip_address.ip_address:
-        AppScaleLogger.log('Azure VM is available at {}'.
-                           format(public_ip_address.ip_address))
-        break
-      AppScaleLogger.verbose("Waiting {} second(s) for IP address to be "
-        "available".format(self.SLEEP_TIME), verbose)
-      time.sleep(self.SLEEP_TIME)
+    sleep_time = self.MAX_VM_CREATION_TIME
+    while sleep_time > 0:
+      try:
+        public_ip_address = network_client.public_ip_addresses.\
+          get(resource_group, vm_network_name)
+        if public_ip_address.ip_address:
+          AppScaleLogger.log('Azure VM is available at {}'.
+                             format(public_ip_address.ip_address))
+          break
+        AppScaleLogger.verbose("Waiting {} second(s) for IP address to be "
+          "available".format(self.SLEEP_TIME), verbose)
+        time.sleep(sleep_time)
+        sleep_time -= self.SLEEP_TIME
```

Comment:
I'm not sure I understand this. It sleeps for less time each loop and the total wait time is `MAX_VM_CREATION_TIME * (MAX_VM_CREATION_TIME / SLEEP_TIME) / 2`? Would there be a way to make it clearer how long this is going to wait for?